src/mbgl/renderer/style_diff.cpp

```diff
@@ -43,14 +38,27 @@ StyleDifference<T> diff(const std::vector<T>& a, const std::vector<T>& b) {
     return result;
 }
 
+ImageDifference diffImages(const std::vector<ImmutableImage>& a,
+                           const std::vector<ImmutableImage>& b) {
+    return diff(a, b, [] (const ImmutableImage& lhs, const ImmutableImage& rhs) {
+        return lhs->id == rhs->id;
```

Comment:
In that case, the image will be [added to the "changed" list](https://github.com/mapbox/mapbox-gl-native/blob/71e46487268cf0eeca3ddbd330d18ec9614990b5/src/mbgl/renderer/style_diff.cpp#L30), and subsequently passed to `SpriteAtlas::addImage`.

* If the image is the same size, this will work.
* If the image is a different size, this will fail an assertion.

#7618 is the underlying issue that should be fixed; I'm planning on working on that this week. In the meantime, I'll change the assertion to a regular conditional, mirroring [this check](https://github.com/mapbox/mapbox-gl-native/blob/71e46487268cf0eeca3ddbd330d18ec9614990b5/src/mbgl/style/style.cpp#L525-L528).